src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -904,16 +904,6 @@ private ClaimsPrincipal ValidateToken(string token, JwtSecurityToken outerToken,
                 }
                 else if (TokenUtilities.IsRecoverableException(exceptionThrown.SourceException))
                 {
-                    if (TokenUtilities.IsRecoverableConfiguration(validationParameters, currentConfiguration, out currentConfiguration))
```

Comment:
Same comment as JsonWebTokenHandler, no need for ReferenceEquals check.

Reply:
updated the logic